File: AED2/TP02/Q02_RegistroC/Series.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

struct Series {
    char name[80];
    char format[80];
    char duration[80];
    char initialCountry[80];
    char originalLang[80];
    char broadcasterInitial[80];
    char streamingInitial[80];
    int numberSeasons;
    int numberEpisodes;
};

void printSerie(struct Series serie) {
    printf("%s %s %s %s %s %s %s %d %d", serie.name, serie.format, serie.duration, serie.initialCountry, serie.originalLang, serie.broadcasterInitial, serie.streamingInitial, serie.numberSeasons, serie.numberEpisodes);
}

void removeTags(char* line) {
    char newLine[80];
    int i = 0, j = 0;
    while (i < strlen(line)) {
        if (line[i] == '<') {
            i++;
            while (line[i] != '>')
                i++;
        }
        else {
            newLine[j] = line[i];
            j++;
        }
        i++;
    }
    while (newLine[j] == ' ' || newLine[j] == '\n')
        j--;
    newLine[--j] == '\0';
    printf("%i\n", newLine[j]);
}

struct Series readData(char* filePath) {
    struct Series serie;

    FILE* fp = fopen(filePath, "r");
    char line[500];


    if (fp == NULL) {
        printf("File not fount.\n");
        exit(EXIT_FAILURE);
    }


    char tmp = ' ';
    int j = 0;
    for (int i = 0; tmp != '\0'; tmp = filePath[++i]) {
        // printf("%d - %c\n", i, tmp);
        if (tmp == '/') {
            j = 0;
        }
        else {
            serie.name[j] = tmp;
            serie.name[j + 1] = '\0';
            j++;
        }
    }
    printf("%s\n", serie.name);

    while (fgets(line, 500, fp)) {
        if (strstr(line, "Formato") != NULL) {
            fgets(line, 500, fp);
            removeTags(line);
            break;
        }
    }

    while (fgets(line, 500, fp)) {
        if (strstr(line, "Duração") != NULL) {
            fgets(line, 500, fp);
            removeTags(line);
            break;
        }
    }

    while (fgets(line, 500, fp)) {
        if (strstr(line, "País de origem") != NULL) {
            fgets(line, 500, fp);
            removeTags(line);
            break;
        }
    }

    while (fgets(line, 500, fp)) {
        if (strstr(line, "Idioma original") != NULL) {
            fgets(line, 500, fp);
            removeTags(line);
            break;
        }
    }

    while (fgets(line, 500, fp)) {
        if (strstr(line, "Emissora de televisão original") != NULL) {
            fgets(line, 500, fp);
            removeTags(line);
        }
    }

    // while (fgets(line, 500, fp)) {
    //     if (strstr(line, "temporadas") != NULL) {
    //         fgets(line, 500, fp);
    //     }
    // }

    // while (fgets(line, 500, fp)) {
    //     if (strstr(line, "episódios") != NULL) {
    //         fgets(line, 500, fp);
    //         printf("%s\n", line);
    //         break;
    //     }
    //     // serie.numberEpisodes = (int)serie.removeTags(file.readLine()).split(" ")[0]);
    // }

    fclose(fp);

    return serie;
}

int main(int argc, char const* argv[]) {
    struct Series seriesList[100];

    char input[80];
    fgets(input, 100, stdin);
    input[strcspn(input, "\r\n")] = 0;

    int count = 0;
    while (strcmp(input, "FIM") != 0) {
        char filePath[100];
        strcpy(filePath, "/tmp/series/");
        strcat(filePath, input);

        seriesList[count] = readData(filePath);

        count++;
        fgets(input, 100, stdin);
        input[strcspn(input, "\r\n")] = 0;
    }

    for (int i = 0; i < count; i++)
        printSerie(seriesList[i]);

    return 0;
}

File: AED2/TP02/Q03_PesquisaSequencial/Series.java
package Q01_ClasseSeries;
import java.io.InputStreamReader;
import java.io.BufferedReader;
import java.io.FileInputStream;
import java.io.Reader;
import java.nio.charset.StandardCharsets;

public class Series {

    private String name;
    private String format;
    private String duration;
    private String initialCountry;
    private String originalLang;
    private String broadcasterInitial;
    private String streamingInitial;
    private int numberSeasons;
    private int numberEpisodes;

    public Series() {

    }

    public Series(String data[]) {
        this.name = data[0];
        this.format = data[1];
        this.duration = data[2];
        this.initialCountry = data[3];
        this.originalLang = data[4];
        this.broadcasterInitial = data[5];
        this.streamingInitial = data[6];
        this.numberSeasons = Integer.parseInt(data[7]);
        this.numberEpisodes = Integer.parseInt(data[8]);
    }

    /** Getters */

    public String getName() {
        return name;
    }

    public String getFormat() {
        return format;
    }

    public String getDuration() {
        return duration;
    }

    public String getInitialCountry() {
        return initialCountry;
    }

    public String getBroadcasterInitial() {
        return broadcasterInitial;
    }

    public int getNumberSeasons() {
        return numberSeasons;
    }

    public int getNumberEpisodes() {
        return numberEpisodes;
    }

    /** Setters */

    public void setName(String name) {
        this.name = name;
    }

    public void setFormat(String format) {
        this.format = format;
    }

    public void setDuration(String duration) {
        this.duration = duration;
    }

    public void setInitialCountry(String initialCountry) {
        this.initialCountry = initialCountry;
    }

    public void setBroadcasterInitial(String broadcasterInitial) {
        this.broadcasterInitial = broadcasterInitial;
    }

    public void setNumberSeasons(int numberSeasons) {
        this.numberSeasons = numberSeasons;
    }

    public void setNumberEpisodes(int numberEpisodes) {
        this.numberEpisodes = numberEpisodes;
    }

    /** Print in screen all data from class */
    public void print() {
        System.out.println(this.name + ' ' + this.format + ' ' + this.duration + ' ' + this.initialCountry + ' '
                + this.originalLang + ' ' + this.broadcasterInitial + ' ' + this.streamingInitial + ' '
                + this.numberSeasons + ' ' + this.numberEpisodes);
    }

    public String removeTags(String line) {
        String newLine = "";
        int i = 0;
        while (i < line.length()) {
            if (line.charAt(i) == '<') {
                i++;
                while (line.charAt(i) != '>')
                    i++;
            } else {
                newLine += line.charAt(i);
            }
            i++;
        }
        System.out.println(newLine);
        return newLine;
    }

    public void readData(String fileName) throws Exception {
        FileInputStream fis = new FileInputStream(fileName);
        InputStreamReader isr = new InputStreamReader(fis, StandardCharsets.UTF_8);
        BufferedReader reader = new BufferedReader(isr);

        // String str;
        // while ((str = reader.readLine()) != null) {
        //     System.out.println(str);
        // }

        while (!reader.readLine().contains("<meta"))
            ;
        this.name = this.removeTags(reader.readLine());

        while (!reader.readLine().contains("Formato"))
            ;

        this.format = this.removeTags(reader.readLine());

        while (!reader.readLine().contains("Dura"))
            ;

        this.duration = this.removeTags(reader.readLine());

        while (!reader.readLine().contains("de origem"))
            ;

        this.initialCountry = this.removeTags(reader.readLine());

        while (!reader.readLine().contains("Idioma original"))
            ;

        this.originalLang = this.removeTags(reader.readLine());

        while (!reader.readLine().contains("de temporadas"))
            ;

        this.numberSeasons = Integer.parseInt(this.removeTags(reader.readLine()));

        while (!reader.readLine().contains("de epis"))
            ;

        this.numberEpisodes = Integer.parseInt(this.removeTags(reader.readLine()));

        reader.close();
    }

    public static void main(String args[]) throws Exception {
        Series serie = new Series();
        serie.readData("series/13_Reasons_Why.html");
        serie.print();
    }

}